pkg/client/cache/reflector.go

```diff
@@ -161,7 +161,8 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 			default:
 				glog.Errorf("Failed to watch %v: %v", r.expectedType, err)
 			}
-			return
+			time.Sleep(time.Second)
```

Comment:
If you catch that error, I'll still be pretty nervous that there's some other unrecoverable error that you're not catching.